packages/frontend/src/redux/slices/transactions/index.js

```diff
@@ -1,22 +1,142 @@
+
+        const { actions: { setTransactions, updateTransactions } } = transactionsSlice;
+
+        !selectTransactionsByAccountId(getState(), { accountId }).length
+            ? dispatch(setTransactions({ transactions, accountId }))
+            : dispatch(updateTransactions({ transactions, accountId }));
+    }
+);
+
+const fetchTransactionStatus = createAsyncThunk(
+    `${SLICE_NAME}/fetchTransactionStatus`,
+    async ({ hash, signer_id, accountId }, { dispatch, getState }) => {
+        let status;
+        try {
+            const transactionDetails = await transactionExtraInfo({ hash, signer_id });
+            status = Object.keys(transactionDetails.status)[0];
+        } catch (error) {
+            status = 'notAvailable';
+        }
+        const checkStatus = ['SuccessValue', 'Failure'].includes(status);
+        const { actions: { updateTransactionStatus } } = transactionsSlice;
+        dispatch(updateTransactionStatus({ status, checkStatus, accountId, hash }));
+    }
+);
+
+const transactionsSlice = createSlice({
+    name: SLICE_NAME,
+    initialState,
+    reducers: {
+        setTransactions(state, { payload }) {
+            const { transactions, accountId } = payload;
+            set(state, ['byAccountId', accountId, 'items'], transactions);
+        },
+        updateTransactions(state, { payload }) {
+            const { transactions, accountId } = payload;
+
+            const transactionsState = state.byAccountId[accountId].items;
+            const hash = transactionsState.map((t) => t.hash_with_index);
+
+            // when updating the transaction, we do not want to replace the entire array, because for some entries the tx status may already be fetched
+            transactions.forEach((t) => {
+                if (!hash.includes(t.hash_with_index)) {
+                    transactionsState.unshift(t);
+                    if (transactionsState.length > 10) {
+                        transactionsState.pop();
+                    }
+                }
+            });
+        },
+        updateTransactionStatus(state, { payload }) {
+            const { status, checkStatus, accountId, hash } = payload;
+
+            const transactionsState = state.byAccountId[accountId].items;
+
+            const transactionEntry = transactionsState.find((t) => t.hash === hash);
+            if (transactionEntry) {
+                Object.assign(transactionEntry, { status, checkStatus});
+            }
+        }
+    },
+    extraReducers: ((builder) => {
+        builder.addCase(fetchTransactions.pending, (state, { meta }) => {
+            const { accountId } = meta.arg;
+
+            set(state, ['byAccountId', accountId, 'status', 'loading'], true);
+        });
+        builder.addCase(fetchTransactions.fulfilled, (state,  { meta }) => {
+            const { accountId } = meta.arg;
+
+            set(state, ['byAccountId', accountId, 'status', 'loading'], false);
+        });
```

Comment:
Should we be resetting `error` state for these cases?  I assume as the very least that we should be for the `fulfilled` case?

Reply:
Done.